### PR TITLE
Add getCurrentSyncs() in ShadowContentResolver

### LINKED ISSUE #7191: Add getCurrentSyncs() in ShadowContentResolver
getCurrentSyncs, which is a static method, is not included in the ShadowContentResolver.

### PR DESCRIPTION
**Pull request by Akshay2131**
Overview

Added getCurrentSyncs() in ShadowContentResolver.
Close #7191 

Proposed Changes

Added getCurrentSyncs() method will return a list of currently active syncs,

### CODE REVIEW

**Comment by Akshay2131** [comment]
@utzcoz and @hoisie could you please review it?

File: shadows/framework/src/main/java/org/robolectric/shadows/ShadowContentResolver.java [hunk 1]
```diff
@@ -530,6 +531,20 @@ protected static boolean isSyncActive(Account account, String authority) {
     return status != null && status.syncRequests > 0;
   }
 
+  @Implementation
+  protected static List<SyncInfo> getCurrentSyncs() {
```
Comment (utzcoz) [inline shadows/framework/src/main/java/org/robolectric/shadows/ShadowContentResolver.java:535]: Please add test for this shadow implementation. We need every PR adds related tests for its Robolectric's implementation changes.

**Review by utzcoz — CHANGES_REQUESTED** [review]

File: shadows/framework/src/main/java/org/robolectric/shadows/ShadowContentResolver.java [hunk 2]
```diff
@@ -530,6 +531,20 @@ protected static boolean isSyncActive(Account account, String authority) {
...
+    List<SyncInfo> list = new ArrayList<>();
+    for (Map.Entry<String, Map<Account, Status>> map : syncableAccounts.entrySet()) {
+      for (Map.Entry<Account, Status> mp : map.getValue().entrySet()) {
```
Comment (utzcoz) [inline shadows/framework/src/main/java/org/robolectric/shadows/ShadowContentResolver.java:538]: Is `map.getValue()` non-null?

**Comment by utzcoz** [comment]
Please read https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue to learn how to link your PR with issue.

File: shadows/framework/src/main/java/org/robolectric/shadows/ShadowContentResolver.java (hunk 1 above)
Comment (hoisie) [reply shadows/framework/src/main/java/org/robolectric/shadows/ShadowContentResolver.java:535]: 👍 on a test, in ShadowContenResolverTest. It can be something simple like having a call to ContentResolver.requestSync and verifying that the result is in ContentResolver.getCurrentSyncs.

Also maybe checking that if no syncs get requested, ContentResolver.getCurrentSyncs returns the empty list.

Comment (hoisie) [reply shadows/framework/src/main/java/org/robolectric/shadows/ShadowContentResolver.java:535]: Also a test that calls requestSync followed by cancelSync, and the value should not be in `ContentResolver.getCurrentSync`.

File: shadows/framework/src/main/java/org/robolectric/shadows/ShadowContentResolver.java (hunk 2 above)
Comment (Akshay2131) [reply shadows/framework/src/main/java/org/robolectric/shadows/ShadowContentResolver.java:538]: Here, `map.getValue()` represents the Map<Account, Status> map.
So, if map is null then map.getValue() can be null.
We can handle this case as: `if (map.getValue() == null) return null;`

File: shadows/framework/src/main/java/org/robolectric/shadows/ShadowContentResolver.java (hunk 1 above)
Comment (Akshay2131) [reply shadows/framework/src/main/java/org/robolectric/shadows/ShadowContentResolver.java:535]: ok thanks, i got it.
i implement it.

Comment (Akshay2131) [reply shadows/framework/src/main/java/org/robolectric/shadows/ShadowContentResolver.java:535]: `
@Test

  public void shouldGetCurrentSyncs() {

    assertThat(ContentResolver.getCurrentSyncs().size()).isEqualTo(0);
    ContentResolver.requestSync(a, AUTHORITY, new Bundle());
    ContentResolver.requestSync(b, AUTHORITY, new Bundle());

    List<SyncInfo> syncs = ContentResolver.getCurrentSyncs();
    assertThat(syncs.size()).isEqualTo(2);

    SyncInfo first = syncs.get(0);
    assertThat(first.account).isEqualTo(a);
    assertThat(first.authority).isEqualTo(AUTHORITY);

    SyncInfo second = syncs.get(1);
    assertThat(second.account).isEqualTo(b);
    assertThat(second.authority).isEqualTo(AUTHORITY);

    ContentResolver.cancelSync(a, AUTHORITY);
    List<SyncInfo> syncsAgain = ContentResolver.getCurrentSyncs();
    assertThat(syncsAgain.size()).isEqualTo(1);

    SyncInfo firstAgain = syncsAgain.get(0);
    assertThat(firstAgain.account).isEqualTo(b);
    assertThat(firstAgain.authority).isEqualTo(AUTHORITY);
  }`

This is the test i have added in ShadowContentResolverTest.java

File: shadows/framework/src/main/java/org/robolectric/shadows/ShadowContentResolver.java (hunk 2 above)
Comment (utzcoz) [reply shadows/framework/src/main/java/org/robolectric/shadows/ShadowContentResolver.java:540]: I think you should write it:
```
if (map.getValue() == null) {
    continue;
}
```

File: robolectric/src/test/java/org/robolectric/shadows/ShadowContentResolverTest.java [hunk 3]
```diff
@@ -641,6 +678,32 @@ public void shouldKnowIfSyncIsActive() {
...
+    assertThat(second.account).isEqualTo(b);
+    assertThat(second.authority).isEqualTo(AUTHORITY);
+
+    ContentResolver.cancelSync(a, AUTHORITY);
+    List<SyncInfo> syncsAgain = ContentResolver.getCurrentSyncs();
+    assertThat(syncsAgain.size()).isEqualTo(1);
+
+    SyncInfo firstAgain = syncsAgain.get(0);
+    assertThat(firstAgain.account).isEqualTo(b);
+    assertThat(firstAgain.authority).isEqualTo(AUTHORITY);
```
Comment (utzcoz) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowContentResolverTest.java:704]: What about adding a case to test cancel b and list is empty?

File: shadows/framework/src/main/java/org/robolectric/shadows/ShadowContentResolver.java [hunk 4]
```diff
@@ -530,6 +531,22 @@ protected static boolean isSyncActive(Account account, String authority) {
     return status != null && status.syncRequests > 0;
   }
 
+  @Implementation
+  protected static List<SyncInfo> getCurrentSyncs() {
+    List<SyncInfo> list = new ArrayList<>();
+    if (syncableAccounts == null) return null;
+    for (Map.Entry<String, Map<Account, Status>> map : syncableAccounts.entrySet()) {
+      if (map.getValue() == null) return null;
```
Comment (utzcoz) [inline shadows/framework/src/main/java/org/robolectric/shadows/ShadowContentResolver.java:539]: Please replace it with:

```Java
if (map.getValue() == null) {
    continue;
}
```

File: shadows/framework/src/main/java/org/robolectric/shadows/ShadowContentResolver.java [hunk 5]
```diff
@@ -530,6 +531,22 @@ protected static boolean isSyncActive(Account account, String authority) {
     return status != null && status.syncRequests > 0;
   }
 
+  @Implementation
+  protected static List<SyncInfo> getCurrentSyncs() {
+    List<SyncInfo> list = new ArrayList<>();
+    if (syncableAccounts == null) return null;
```
Comment (utzcoz) [inline shadows/framework/src/main/java/org/robolectric/shadows/ShadowContentResolver.java:537]: Please add `{}` for it:

```Java
if (syncableAccounts == null) {
    return null;
}
```

File: shadows/framework/src/main/java/org/robolectric/shadows/ShadowContentResolver.java [hunk 6]
```diff
@@ -530,6 +531,22 @@ protected static boolean isSyncActive(Account account, String authority) {
...
+    for (Map.Entry<String, Map<Account, Status>> map : syncableAccounts.entrySet()) {
+      if (map.getValue() == null) continue;
```
Comment (utzcoz) [inline shadows/framework/src/main/java/org/robolectric/shadows/ShadowContentResolver.java:539]:
```Java
if (map.getValue() == null) {
    continue;
}
```

File: shadows/framework/src/main/java/org/robolectric/shadows/ShadowContentResolver.java [hunk 7]
```diff
@@ -530,6 +531,26 @@ protected static boolean isSyncActive(Account account, String authority) {
     return status != null && status.syncRequests > 0;
   }
 
+  @Implementation
+  protected static List<SyncInfo> getCurrentSyncs() {
+    List<SyncInfo> list = new ArrayList<>();
+    if (syncableAccounts == null) {
+      return null;
```
Comment (utzcoz) [inline shadows/framework/src/main/java/org/robolectric/shadows/ShadowContentResolver.java:538]: Oh, I remember the result of this method can't be null. Could you check official documentation of this method?

Comment (Akshay2131) [reply shadows/framework/src/main/java/org/robolectric/shadows/ShadowContentResolver.java:538]: Yeah, syncableAccounts==null is always going to false.
I think we should remove it from the code.
It wouldn't impact the rest of code.

Comment (Akshay2131) [reply shadows/framework/src/main/java/org/robolectric/shadows/ShadowContentResolver.java:538]: I removed it and now it's working fine.
Can i finally commit again?

**Review by utzcoz — APPROVED** [review]
Thanks @Akshay2131 . LGTM. cc @hoisie.

**Comment by Akshay2131** [comment]
Thanks for your support @utzcoz and @hoisie

**Comment by utzcoz** [comment]
Hi @hoisie , could you help to review this PR? Thanks.

**Review by hoisie — APPROVED** [review]

**Comment by hoisie** [comment]
Not a big deal, but in general I like to see formatting changes in separate commits, as it keeps git commit history cleaner.

**Comment by utzcoz** [comment]
> Not a big deal, but in general I like to see formatting changes in separate commits, as it keeps git commit history cleaner.

I prefer to combine formatting with related changed, because code change maybe bring new formatting problem. If separate commits for formatting are acceptable, I will not request contributors to do it at later review progress.
